lib/learn_to_rank/ctr_to_judgements.rb

```diff
@@ -0,0 +1,67 @@
+module LearnToRank
+  class CtrToJudgements
+    # CtrToJudgements takes an array of query:position click-through-rates
+    # and turns them into normalised relevancy judgements between 0 and 3.
+    # INPUT [{
+    #   "my query" => {
+    #     "1": 10,
+    #     "2": 23,
+    #     "3": 40,
+    #     "4": 5,
+    #   }
+    # }]
+    # OUTPUT [{ query: 'my query', score: 3, link: '/the-best-result' }]
+    def initialize(ctrs)
+      @ctrs = ctrs.reduce({}, :merge)
+    end
+
+    def relevancy_judgements
+      ctrs.map { |(query, query_ctrs)|
+          sleep 0.1
+          scores_to_results(query, ctr_to_relevancy_score(query_ctrs))
+        }
+        .flatten
+        .compact
+    end
+
+  private
+
+    attr_reader :ctrs
+
+    def ctr_to_relevancy_score(query_ctrs)
```

Comment:
Did the refinement I suggest end up not working out? I could well believe it!

Reply:
Have you tried it in your PR? https://github.com/alphagov/search-api/pull/1775